### PR TITLE
CIRC-395: resolve request queue positions issue

### PR DESCRIPTION
**Pull request by wwelling**
## Purpose

Resolves [CIRC-395](https://issues.folio.org/browse/CIRC-395) [issue](https://issues.folio.org/browse/CIRC-395?focusedCommentId=55779&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-55779) discovered during testing. 

## Approach
- reordering of actions to move request
- reorder request queue on newly created request
- updating requests in order which avoid duplicate positions in request queue
- additional test for specific scenario of moving two requests from one item to another
- cherry-pick excellent work by @marc-johnson on adding validation to the request test fixture
- update tests

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] Were any API paths or methods changed, added or removed? No
  - [x] Were there any schema changes? No
  - [x] Did any of the interface versions change? No
  - [x] Were permissions changed, added, or removed? No
  - [x] Are there new interface dependencies? No
  - [x] There are no breaking changes in this PR. No

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/RequestQueue.java [hunk 1]
```diff
@@ -48,18 +49,17 @@ public boolean isRequestedByOtherPatron(User requestingUser) {
...
   }
 
   public void remove(Request request) {
-    // NOTE: tests use Arrays.asList which produces a collection which
-    // does not support remove
     requests = requests.stream()
       .filter(r -> !r.getId().equals(request.getId()))
       .collect(Collectors.toList());
     request.removePosition();
+    request.freePreviousPosition();
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/RequestQueue.java]: What is the value in rem Bering the previous position on removal, when it is forgotten immediately afterwards?

File: src/main/java/org/folio/circulation/domain/RequestQueue.java [hunk 2]
```diff
@@ -86,9 +86,17 @@ public Boolean contains(Request request) {
   Collection<Request> getRequestsWithChangedPosition() {
     return requests.stream()
       .filter(Request::hasChangedPosition)
+      // order by position descending
+      .sorted((req1, req2) -> req2.getPosition().compareTo(req1.getPosition()))
       .collect(Collectors.toList());
   }
 
+  public Boolean positionTaken(Request request) {
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/RequestQueue.java]: This seems to check if the position was previously taken by another request. Is that a reasonable understanding?
```suggestion
  public Boolean positionPreviouslyTaken(Request request) {
```

File: src/main/java/org/folio/circulation/domain/RequestQueue.java (hunk 1 above)
Comment (wwelling) [reply src/main/java/org/folio/circulation/domain/RequestQueue.java]: There is no value. Forgot to remove it. Thanks.

File: src/main/java/org/folio/circulation/domain/RequestQueue.java (hunk 2 above)
Comment (wwelling) [reply src/main/java/org/folio/circulation/domain/RequestQueue.java]: Yes, that is correct. I like the name suggestion. Will update.

**Pull request merged**